pkg/kubelet/status/generate.go

```diff
@@ -154,12 +154,19 @@ func GeneratePodInitializedCondition(spec *v1.PodSpec, containerStatuses []v1.Co
 			Reason: UnknownContainerStatuses,
 		}
 	}
+
 	unknownContainers := []string{}
-	unreadyContainers := []string{}
+	incompleteContainers := []string{}
 	for _, container := range spec.InitContainers {
 		if containerStatus, ok := podutil.GetContainerStatus(containerStatuses, container.Name); ok {
-			if !containerStatus.Ready {
-				unreadyContainers = append(unreadyContainers, container.Name)
+			if kubetypes.IsRestartableInitContainer(&container) {
+				if containerStatus.Started == nil || !*containerStatus.Started {
+					incompleteContainers = append(incompleteContainers, container.Name)
+				}
+			} else { // regular init container
+				if !containerStatus.Ready {
+					incompleteContainers = append(incompleteContainers, container.Name)
+				}
```

Comment:
I think this would be more readable and cleaner refactored into a seperate method, maybe `isInitContainerInitialized` that would:

(1) Check if it is a IsRestartableInitContainer or not
(2) depending of it is or not, check containerStatus.Started vs containerStatus.Ready

So something like

```
func isInitContainerInitialized(initContainer *v1.Container, containerStatus *v1.ContainerStatus) bool {
    if kubetypes.IsRestartableInitContainer(&container) {
        if containerStatus.Started == nil || !*containerStatus.Started {
            incompleteContainers = append(incompleteContainers, container.Name)
        }
    } else { // regular init container
        if !containerStatus.Ready {
            incompleteContainers = append(incompleteContainers, container.Name)
        }
    }
}
```

Then call the helper function here. I think that would clean up the multiple levels of if indentation and make the intent a bit more clear.

Reply:
sounds good. refactored it.